storybookjs/storybook
# Controls: Fix UI to add array items

Closes #22903

## What I did

I applied the same style to `.rejt-array-node > span > *` as `.rejt-object-node > span > *`

The bug was caused by the plus button being not clickable because it is obscured by the pseudo element in the same level (`.rejt-not-collapsed::before`), which is absolutely positioned with z-index 1. So by applying `position: relative` and `z-index: 2` to the selector `.rejt-array-node > span > *`, the plus button can now be interacted if the object is an array as well.

## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Add some test array to the primary story in `sandbox/react-vite-default-ts/src/stories/Button.stories.ts`
```ts
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Primary: Story = {
  args: {
    primary: true,
    label: 'Button',
    array: [{
      a: 1,
      b: "2",
      c: false
    }, {
      a: 2,
      b: "3",
      c: true
    }]
  },
};
```
3. Go over to Controls panel for Primary story and look for `array` field.
4. Interact with the plus button in the first line and see if it works.

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
